### PR TITLE
chore: Make meeting series naming consistent

### PR DESCRIPTION
**Pull request by Dschoordsch**
Fixes #9928

Make the meeting name a separate parameter so it can be used for the meeting, meeting series and gcal event. This makes everything more consistent.

## Demo

https://www.loom.com/share/63b49765c63b4575a59838ab4fc8a402?sid=5fd46150-274e-4b70-baad-08d12521516a

## Testing scenarios

[Please list all the testing scenarios a reviewer has to check before approving the PR]

- [ ] Scenario A
  - Step 1
  - Step 2...

- [ ] Scenario B
  - Step 1
  - Step 2....

## Final checklist

- [ ] I checked the [code review guidelines](../docs/codeReview.md)
- [ ] I have added [Metrics Representative](../docs/codeReview.md#metrics-representative) as reviewer(s) if my PR invovles metrics/data/analytics related changes
- [ ] I have performed a self-review of my code, the same way I'd do it for any other team member
- [ ] I have tested all cases I listed in the testing scenarios and I haven't found any issues or regressions
- [ ] Whenever I took a non-obvious choice I added a comment explaining why I did it this way
- [ ] I added the label https://github.com/ParabolInc/parabol/labels/Skip%20Maintainer%20Review if the PR introduces only minor changes, does not contain any architectural changes or does not introduce any new patterns and I think one review is sufficient'
- [ ] PR title is human readable and could be used in changelog

## Summary by CodeRabbit

- **New Features**
  - Introduced `name` and `rrule` parameters for scheduling and starting activities.
  - Added a function to calculate the count of meetings based on team ID and meeting type.

- **Refactor**
  - Replaced `RecurrenceSettingsInput` with `name` and `rrule` parameters across various components and mutations.
  - Simplified the structure of recurrence settings in updates and mutations.
  - Adjusted imports and function signatures accordingly.

- **Bug Fixes**
  - Updated default name for retrospectives to 'Retro #1'.
  - Added a new test for naming a single retrospective.

### CODE REVIEW

File: packages/server/graphql/mutations/helpers/safeCreateRetrospective.ts [hunk 1]
```diff
@@ -16,29 +16,22 @@ const safeCreateRetrospective = async (
     videoMeetingURL?: string
     meetingSeriesId?: number
     scheduledEndTime?: Date
-    name?: string
+    name: string
```
[the remark this answers is not in the document]
Comment (Dschoordsch) [reply packages/server/graphql/mutations/helpers/safeCreateRetrospective.ts:19]: You're wrong, it's clearly there: https://github.com/ParabolInc/parabol/blob/0fb3d7e5db8c28c08ddeb1df3f251c2d187e2c43/packages/server/graphql/private/mutations/processRecurrence.ts#L96

[the remark this answers is not in the document]
Comment (Dschoordsch) [reply packages/server/graphql/mutations/helpers/safeCreateRetrospective.ts:19]: Your context is too small, if you run the script
```
rg --type js --type ts --line-number --context 12 $'const meeting = await safeCreateRetrospective' packages/server/graphql/private/mutations/processRecurrence.ts
```
you will realize that the parameter is already present.

[the remark this answers is not in the document]
Comment (mattkrick) [reply packages/server/graphql/mutations/helpers/safeCreateRetrospective.ts:19]: 🤣 i'm DYING. i don't know why i find this so hilarious, but i love it.

File: packages/server/dataloader/customLoaderMakers.ts [hunk 2]
```diff
@@ -889,3 +889,27 @@ export const fileStoreAsset = (parent: RootDataLoader) => {
...
+        })
+      )
+      return res
+    },
+    {
+      ...parent.dataLoaderOptions,
+      cacheKeyFn: (key) => `${key.teamId}:${key.meetingType}`
+    }
+  )
+}
```
[the remark this answers is not in the document]
Comment (Dschoordsch) [reply packages/server/dataloader/customLoaderMakers.ts:915]: We don't want to catch an error only to log and rethrow it. The error will bubble up and get logged automatically. Logging it in addition just spams the logs and messes with the backtrace.

**Pull request merged**